### ISSUE TITLE
Keep training tera if the initial maximum step is not enough

### ISSUE THREAD

**Issue by toco2270853** [s3prl/s3prl#180]
Dear developer, I'm grateful for the provided repo. As the title, I have train
a tera model, but consider resource limitation, I restrict total training epoch to 
about 115 on LibriSpeech 100 hour subset with batch size 8. But I found the result is a bit lower than one report by the paper, and I think it's due to the small batch size and smaller total epoch(but I have run more step than paper, I'm not sure whether running more step will get better result comparing to more epoch). I don't want to train the model again by reset the maximum training step and retrain. But tera train the model with learning rate linearly decay to 0 at maximum training step. I don't know how to keep training it. I'm wondering is there any good way for solving this problem?

**Comment by andi611** [comment]
Hi, currently we do not support this function.

However, there is a quick workaround:
At the beginning of the pre-training script, manually add a few line of code to load your pre-trained TERA model, and continue the pre-training script like before (you can adjust the optimizer settings in the config file to suit your need). I believe this will solve your problem.

**Comment by toco2270853** [comment]
Thanks for the quick reply. I really appreciate it!
 If I didn't misunderstand your advice. If I load the pretrain
model first, and start training from script again(assume I use the same script, so it's equivalent to double the training epoch, and the learning rate will cycle for two cycles.)
My experience about training the transformer is, it may be sensitive to learning rate scheduling process. For example, if I didn't use enough warm up step, then the training may crash sometimes. It's there any paper using this kind of learning rate scheduling process(multiple cycle)?May I ask the title of those papers?

**Comment by andi611** [comment]
Hi, I wouldn't suggest using two cycles. Instead you can modify the optimizer to have no schedule.
Change this [line](https://github.com/s3prl/s3prl/blob/558647a69fcc84f69fa6b5524c6e999c76118f9a/s3prl/pretrain/tera/config_runner.yaml#L12) to something else, for example [this optimizer](https://github.com/s3prl/s3prl/blob/558647a69fcc84f69fa6b5524c6e999c76118f9a/s3prl/optimizers.py#L80) by changing the string to "AdamW".
You can also use a different number of epoch on the second attempt.
